Inductor cpp wrapper: support dynamic shapes

chunyuan-w opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #97965


1. Fixed dynamic shapes support in cpp_wrapper
   - fixed the cpp codegen of `size()` and `stride()`
   - fixed the cpp codegen of `ShapeAsConstantBuffer`
   - changed to use `cexpr` instead of `pexpr` in the cpp codegen of the `sizevar`


2. Enabled dynamic shapes tests for cpp_wrapper

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

chunyuan-w committed:
Inductor cpp wrapper: support dynamic shapes

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"



1. Fixed dynamic shapes support in cpp_wrapper
   - fixed the cpp codegen of `size()` and `stride()`
   - fixed the cpp codegen of `ShapeAsConstantBuffer`
   - changed to use `cexpr` instead of `pexpr` in the cpp codegen of the `sizevar`


2. Enabled dynamic shapes tests for cpp_wrapper

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"



1. Fixed dynamic shapes support in cpp_wrapper
   - fixed the cpp codegen of `size()` and `stride()`
   - fixed the cpp codegen of `ShapeAsConstantBuffer`
   - changed to use `cexpr` instead of `pexpr` in the cpp codegen of the `sizevar`


2. Enabled dynamic shapes tests for cpp_wrapper

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"



1. Fixed dynamic shapes support in cpp_wrapper
   - fixed the cpp codegen of `size()` and `stride()`
   - fixed the cpp codegen of `ShapeAsConstantBuffer`
   - changed to use `cexpr` instead of `pexpr` in the cpp codegen of the `sizevar`


2. Enabled dynamic shapes tests for cpp_wrapper

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"



1. Fixed dynamic shapes support in cpp_wrapper
   - fixed the cpp codegen of `size()` and `stride()`
   - fixed the cpp codegen of `ShapeAsConstantBuffer`
   - changed to use `cexpr` instead of `pexpr` in the cpp codegen of the `sizevar`


2. Enabled dynamic shapes tests for cpp_wrapper

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jgong5 commented on test/inductor/test_torchinductor.py:
```diff
@@ -1669,6 +1669,25 @@ def fn(a, b):
+
+        self.common(fn, [torch.randint(5, (1, 8)), 5400])
+
+    def test_int_div(self):
+        if self.device == "cuda":
+            raise unittest.SkipTest("only support cpu int_div test")
```
If so, consider to move it to CPURepro?

chunyuan-w replied:
Moved to `CPURepro`.

jgong5 reviewed (approved):
LGTM. Suggest to move CPU-only test to the right place. See comment.

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"



1. Fixed dynamic shapes support in cpp_wrapper
   - fixed the cpp codegen of `size()` and `stride()`
   - fixed the cpp codegen of `ShapeAsConstantBuffer`
   - changed to use `cexpr` instead of `pexpr` in the cpp codegen of the `sizevar`


2. Enabled dynamic shapes tests for cpp_wrapper

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"



1. Fixed dynamic shapes support in cpp_wrapper
   - fixed the cpp codegen of `size()` and `stride()`
   - fixed the cpp codegen of `ShapeAsConstantBuffer`
   - changed to use `cexpr` instead of `pexpr` in the cpp codegen of the `sizevar`


2. Enabled dynamic shapes tests for cpp_wrapper

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support dynamic shapes"



1. Fixed dynamic shapes support in cpp_wrapper
   - fixed the cpp codegen of `size()` and `stride()`
   - fixed the cpp codegen of `ShapeAsConstantBuffer`
   - changed to use `cexpr` instead of `pexpr` in the cpp codegen of the `sizevar`


2. Enabled dynamic shapes tests for cpp_wrapper

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w commented:
@pytorchbot merge
